databricks/koalas/series.py

```diff
@@ -4518,6 +4519,86 @@ def repeat(self, repeats: int) -> "Series":
         else:
             return _col(ks.concat([kdf] * repeats))
 
+    def asof(self, where):
+        """
+        Return the last row(s) without any NaNs before `where`.
+
+        The last row (for each element in `where`, if list) without any
+        NaN is taken.
+
+        If there is no good value, NaN is returned.
+
+        Parameters
+        ----------
+        where : index or array-like of indices
+
+        Returns
+        -------
+        scalar or Series
+
+            The return can be:
+
+            * scalar : when `self` is a Series and `where` is a scalar
+            * Series: when `self` is a Series and `where` is an array-like
+
+            Return scalar or Series
+
+        Notes
+        -----
+        Indices are assumed to be sorted. Raises if this is not the case.
+
+        Examples
+        --------
+        >>> s = ks.Series([1, 2, np.nan, 4], index=[10, 20, 30, 40])
+        >>> s
+        10    1.0
+        20    2.0
+        30    NaN
+        40    4.0
+        Name: 0, dtype: float64
+
+        A scalar `where`.
+
+        >>> s.asof(20)
+        2.0
+
+        For a sequence `where`, a Series is returned. The first value is
+        NaN, because the first element of `where` is before the first
+        index value.
+
+        >>> s.asof([5, 20]).sort_index()
+        5     NaN
+        20    2.0
+        Name: 0, dtype: float64
+
+        Missing values are not considered. The following is ``2.0``, not
+        NaN, even though NaN is at the index location for ``30``.
+
+        >>> s.asof(30)
+        2.0
+        """
+        should_return_series = True
+        if isinstance(self.index, ks.MultiIndex):
+            raise ValueError("asof is not supported for a MultiIndex")
+        if isinstance(where, ks.DataFrame):
+            raise ValueError("where cannot be a DataFrame")
+        if not self.index.is_monotonic_increasing:
```

Comment:
Index.asof checks `is_monotonic_decreasing` too. should we check here too?

Reply:
We don't need to check because pandas only allows the Series with monotonically increasing Index for `Series.asof()` unlike `Index.asof()`.

```python
>>> pd.Index([4, 3, 2]).asof(1)
2

>>> pd.Series([1, 2, 3], index=[4, 3, 2]).asof(1)
...
ValueError: asof requires a sorted index
```